tests/client/test_dataset.py

```diff
@@ -802,6 +804,7 @@ def test_to_from_datasets(self, text2text_records):
             "status",
             "event_timestamp",
             "metrics",
+            "projection",
```

Comment:
Regarding the flow, maybe it makes sense to have this field in HF structure for those cases where projection was applied. Just to prevent reloading data from a projected HF snapshot.

So, we could keep it, but maybe only when `projection=True`